test/nodes/test_retriever.py

```diff
@@ -310,6 +325,9 @@ def test_retriever_basic_search(document_store, retriever, docs_with_ids):
 @pytest.mark.parametrize("document_store", ["elasticsearch", "memory"], indirect=True)
 @pytest.mark.embedding_dim(512)
 def test_table_text_retriever_embedding(document_store, retriever, docs):
+    # BM25 representation is incompatible with table retriever
+    if isinstance(document_store, InMemoryDocumentStore):
+        document_store.use_bm25 = False
```

Comment:
Sorry to drag this PR along a bit more :sweat_smile: but this is a great point! BM25 is not compatible with tables, or any other document type than `text` to be fair. Let's add a check in `write_documents` that, if it detects you're writing non-text documents with `use_bm25=True`, throws a warning telling you that such documents will not be indexed by BM25, but still written in.